node/pkg/watchers/sui/watcher.go

```diff
@@ -84,62 +133,57 @@ func NewWatcher(
 	}
 }
 
-func (e *Watcher) inspectBody(logger *zap.Logger, body gjson.Result) error {
-	txDigest := body.Get("txDigest")
-	timestamp := body.Get("timestamp")
-	moveEvent := body.Get("event.moveEvent")
-	if !moveEvent.Exists() {
-		return nil
+func (e *Watcher) inspectBody(logger *zap.Logger, body SuiResult) error {
+	if (body.Timestamp == nil) || (body.TxDigest == nil) {
+		return errors.New("Missing event fields")
 	}
-	packageId := moveEvent.Get("packageId")
-	account := moveEvent.Get("sender")
-	fields := moveEvent.Get("fields")
-	if !fields.Exists() {
+	if body.Event.MoveEvent == nil {
 		return nil
 	}
-	consistency_level := fields.Get("consistency_level")
-	nonce := fields.Get("nonce")
-	payload := fields.Get("payload")
-	sender := fields.Get("sender")
-	sequence := fields.Get("sequence")
-	if !payload.Exists() {
-		return nil
+	moveEvent := *body.Event.MoveEvent
+	if (moveEvent.PackageID == nil) || (moveEvent.Sender == nil) {
+		return errors.New("Missing event fields")
 	}
 
-	if !txDigest.Exists() || !timestamp.Exists() || !packageId.Exists() || !account.Exists() || !consistency_level.Exists() || !nonce.Exists() || !sender.Exists() || !sequence.Exists() {
-		return errors.New("Missing event fields")
+	if moveEvent.Fields == nil {
+		return nil
+	}
+	fields := *moveEvent.Fields
+	if (fields.ConsistencyLevel == nil) || (fields.Nonce == nil) || (fields.Payload == nil) || (fields.Sender == nil) || (fields.Sequence == nil) {
+		return nil
```

Comment:
Should this be an error since some fields are missing?

Reply:
yes but...  move generates a lot of automatic events as well.  For example, whenever you mutate an account, create a new account, etc.  I was thinking I could add another field into the event saying "THIS is the droid you are looking for" and error if THAT event is missing fields as apposed to how it currently works where it looks for any event that contains all the fields we are looking for.   This is what I did in the near contract.   I will make an issue to modify the contracts and then update the watcher to match